cpp/dolfinx/io/xdmf_function.cpp

```diff
@@ -88,98 +56,153 @@ void xdmf_function::add_function(MPI_Comm comm, const fem::Function<T, U>& u,
   assert(u.function_space());
   auto mesh = u.function_space()->mesh();
   assert(mesh);
+  auto element = u.function_space()->element();
+  assert(element);
+
+  // FIXME: is the below check adequate for detecting a Lagrange
+  // element?
+  // Check that element is Lagrange
```

Comment:
I don't think so. Maybe @mscroggs  has examples of elements with point evaluations that are not Lagrange?